lifecycle-operator/controllers/lifecycle/keptnapp/controller_test.go

```diff
@@ -236,6 +301,40 @@ func TestKeptnAppReconciler_deprecateAppVersions(t *testing.T) {
 	require.Equal(t, apicommon.StateDeprecated, keptnappversion.Status.Status)
 }
 
+func TestKeptnAppReconciler_deprecateAppVersionsError(t *testing.T) {
+
+	app := testcommon.GetApp("myapp")
+	app.Spec.Revision = uint(2)
+	app.Generation = int64(2)
+	appVersion := &lfcv1beta1.KeptnAppVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "myapp-1.0.0-6b86b273",
+			Namespace: "default",
+		},
+	}
+	r, _ := setupReconciler(app, appVersion)
+
+	fakeClient := k8sfake.NewClientBuilder().WithScheme(scheme.Scheme).WithInterceptorFuncs(interceptor.Funcs{
+		Get: func(ctx context.Context, client client.WithWatch, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+			if key.Name == "myapp-1.0.0-6b86b273" {
+				return errors.New("unexpected error")
+			}
+			return client.Get(ctx, key, obj, opts...)
+		},
+	}).WithObjects(app, appVersion).Build()
+
+	r.Client = fakeClient
+
+	_, err := r.Reconcile(context.TODO(), ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: "default",
+			Name:      "myapp",
+		},
+	})
+
+	require.NotNil(t, err)
```

Comment:
let's add more assertions here, we should expect a exact error here. Asserting for some error is little bit too general and can lead to a non-expected error and bugs